src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,305 @@
+                var subscriptionId = subscriptionAndPerformanceConnectionStatsList.Handler.SubscriptionId;
+                var performance = subscriptionAndPerformanceConnectionStatsList.Performance;
+                
+                var connectionsAggregators = new List<SubscriptionConnectionStatsAggregator>(performance.Count);
+
+                // 1. get 'closed' connections info
+                while (performance.TryTake(out SubscriptionConnectionStatsAggregator stat))
+                {
+                    connectionsAggregators.Add(stat);
+                }
+                
+                // 2. get 'inProgress' connection info
+                using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var inProgressConnection = Database.SubscriptionStorage.GetSubscriptionConnection(context, subscriptionName);
+                    var inProgressStats = inProgressConnection?.Connection?.GetPerformanceStats();
+                    
+                    if (inProgressStats != null &&
+                        inProgressStats.Completed == false &&
+                        connectionsAggregators.Contains(inProgressStats) == false)
+                    {
+                       connectionsAggregators.Add(inProgressStats);
+                    }
+                }
+                
+                // 3. add to results
+                connectionPerformance.AddRange(connectionsAggregators.Select(x => x.ToConnectionPerformanceLiveStatsWithDetails()));
+                preparedStats.Add(new SubscriptionTaskPerformanceStats
+                {
+                    TaskName = subscriptionName,
+                    TaskId = subscriptionId,
+                    ConnectionPerformance = connectionPerformance.ToArray()
+                });
+            }
+              
+            foreach (var kvp in _perSubscriptionBatchStats)
+            {
+                var batchPerformance = new List<SubscriptionBatchPerformanceStats>();
+                
+                var subscriptionAndPerformanceBatchStatsList = kvp.Value;
+                var subscriptionName = subscriptionAndPerformanceBatchStatsList.Handler.SubscriptionName;
+                var performance = subscriptionAndPerformanceBatchStatsList.Performance;
+
+                var batchAggregators = new List<SubscriptionBatchStatsAggregator>(performance.Count);
+                
+                // 1. get 'ended' batches info
+                while (performance.TryTake(out SubscriptionBatchStatsAggregator stat))
+                {
+                    batchAggregators.Add(stat);
+                }
+                
+                // 2. get 'inProgress' batch info
+                using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
```

Comment:
db or server context?

Reply:
It's okay. It needs to open read transaction on server store.